src/StorageSync/StorageSync/Interop/Clients/SyncServerRegistrationClientBase.cs

```diff
@@ -141,14 +166,28 @@ public void Dispose()
             uint certificateKeyLength,
             string monitoringDataPath,
             string agentVersion,
+            string serverMachineName,
             Func<string,string,ServerRegistrationData, RegisteredServer> registerOnlineCallback)
         {
+            // Discover the server type , Get the application id, 
+            Guid? applicationId = GetApplicationIdOrNull();
+
+            // Honor identity if present
+            bool isCertificateRegistration = true;// applicationId.GetValueOrDefault(Guid.Empty) == Guid.Empty;
```

Comment:
why is this hardcoded?

Reply:
// We allow only server registration with server certificate in v19.

Reply:
So to achieve MI in v19, you have to create the topology with cert, then run the migration cmdlet? You cannot do it directly from scratch, if I am interpreting correctly?